src/Pages/PeoplePage.tsx

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { Link, useSearchParams } from 'react-router-dom';
+import { useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { PeopleContex } from '../store/PeopleContex';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { getPeople } from '../api';
+import { filterPeople } from '../services/filter';
+import { getSearchWith } from '../utils/searchHelper';
+import { SortType } from '../types/SortType';
+
+export const PeoplePage = () => {
+  const {
+    people,
+    errorMassage,
+    loading,
+    setLoading,
+    setPeople,
+    setErrorMassage,
+  } = useContext(PeopleContex);
+
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const filteredPeople =
+    people && filterPeople(people, query, centuries, sex, sort, order);
```

Comment:
You don't need to check the `people` because your `people` variable is always an array (even when people don't exist at all, this variable will be an empty array).